src/reducetensor.cpp

```diff
@@ -1031,6 +1032,10 @@ void ReduceTensorDescriptor::ReduceTensor(const Handle& handle,
                                        std::to_string(static_cast<int>(use_padding.first)) +
                                        std::to_string(static_cast<int>(use_padding.second));
 
+        int alignment            = 64;
+        Data_t workspace_aligned = std::align(
+            alignment, workspaceSizeInBytes - alignment, workspace, workspaceSizeInBytes);
```

Comment:
[Recommendation] `std::align` changes the value of `workspaceSizeInBytes`, so we should not use its value after this (or use it with caution). Well, actually `workspaceSizeInBytes_aligned` gets has different semantics after this call, something alike `workspaceSizeInBytes_aligned`.

This preserves `workspaceSizeInBytes` and introduces explicit "aligned" kind:

```suggestion
        size_t workspaceSizeInBytes_aligned = workspaceSizeInBytes;
        Data_t workspace_aligned = std::align(
            alignment, workspaceSizeInBytes - alignment, workspace, workspaceSizeInBytes_aligned);
```

Or it may be even better to directly update `workspace`, so the semantically bound pair {`workspace`, `workspaceSizeInBytes`} remains in sync after this statement:

```suggestion
        workspace = std::align(alignment, workspaceSizeInBytes - alignment, workspace, workspaceSizeInBytes);
```

Reply:
@atamazov  `std::align`, on success, changes the value of the `workspace` pointer and decreases space argument by the number of bytes used for alignment. Thus, the value obtained from std::align is either 0 at fail or identical to new state of `workspace` at success.